### DISCUSSION TITLE
Can't upload file with `multipart/form-data` in Deno Deploy

### DISCUSSION THREAD

**Discussion by jcs224** [oakserver/oak#595] [Q&A]
This app works fine with the Deno CLI, but not Deno Deploy:

https://dash.deno.com/playground/jcs224-zbar-wasm-oak

The `multipart/form-data` file upload doesn't work because it's trying to run `Deno.makeTempDir` and Deploy doesn't have the ability to write files.

The failing line is 27, `const data = await body.value.read()`.

Is there a way to get the multipart file data without writing a temp dir? I was able to do this with the web-native `FormData` without Oak:

https://developer.mozilla.org/en-US/docs/Web/API/FormData/get

**Comment by jcs224** [comment]
Or, at least, is there a way to access the underlying `Request` object in Oak, within a route? I'm struggling to find it if there is...

**Comment by yubin-code** [comment]
Have you found a solution

**Comment by jcs224** [comment]
Actually, I have! Feels a little hack-y, but it works. It involves getting the raw request payload, and parsing it ourselves.

Here's a quick snippet, not a full example:
```ts
// Using a multiparser elsewhere in the community
import { multiParser } from 'https://deno.land/x/multiparser/mod.ts'
// ...
router
.post('/upload', async (ctx, next) => {
  const body = await ctx.request.body({
    type: 'form-data'
  })

  const form = await multiParser(ctx.request.originalRequest.request)

  const img = form.files['image-upload'].content
  // ...
})
```

Here's a complete example: https://dash.deno.com/playground/jcs224-oak-zbar-multipart

I still think this should be fixed within Oak, but this is a good enough workaround for now.

**Comment by linhub15** [comment]
The type `form-data` writes to file because the `maxSize` defaults to `0`.

https://github.com/oakserver/oak/blob/91b7aaa1d0bc7b20cab50cc6ddf7a1b9a7d0b567/multipart.ts#L111-L118

Try setting a `maxSize` value to a number larger than the expected image size (in bytes).
```
const data = await body.value.read({ maxSize: 10_000_000 });
```
